lib/xcode/install.rb

```diff
@@ -58,16 +59,31 @@ def current_symlink
     end
 
     def download(version, progress, url = nil)
-      return unless url || exist?(version)
-      xcode = seedlist.find { |x| x.name == version } unless url
+      xcode = find_xcode_version(version) if url.nil?
+      return if url.nil? && xcode.nil?
+
       dmg_file = Pathname.new(File.basename(url || xcode.path))
 
       result = Curl.new.fetch(url || xcode.url, CACHE_DIR, url ? nil : spaceship.cookie, dmg_file, progress)
       result ? CACHE_DIR + dmg_file : nil
     end
 
+    def find_xcode_version(version)
+      # By checking for the name and the version we have the best success rate
+      # Sometimes the user might pass
+      #   "4.3 for Lion"
+      # or they might pass an actual Gem::Version
+      #   Gem::Version.new("8.0.0")
+      # which should automatically match with "Xcode 8"
+      seedlist.each do |current_seed|
+        return current_seed if current_seed.version == Gem::Version.new(version)
+        return current_seed if current_seed.name == version
+      end
+      nil
+    end
+
     def exist?(version)
-      list_versions.include?(version)
+      !find_xcode_version(version).nil?
```

Comment:
this is confusing to read, I'd consider using a temporary variable and then nil checking that.